store/views.py

```diff
@@ -1,82 +1,263 @@
+    # Search for Loaned Books by the user
+    books = BookCopy.objects.filter(borrower = request.user)
     context = {
-        'books': None,
+        'books': books,
     }
-    '''
-    The above key 'books' in the context dictionary should contain a list of instances of the 
-    BookCopy model. Only those book copies should be included which have been loaned by the user.
-    '''
-    # START YOUR CODE HERE
-    
-
-
     return render(request, template_name, context=context)
 
 @csrf_exempt
 @login_required
 def loanBookView(request):
-    response_data = {
-        'message': None,
-    }
     '''
     Check if an instance of the asked book is available.
     If yes, then set the message to 'success', otherwise 'failure'
     '''
-    # START YOUR CODE HERE
-    book_id = None # get the book id from post data
+    if request.method == 'POST':
+        # Get the copies of the required book if it is available for loan
+        book_id = request.POST['bid']
+        books = BookCopy.objects.filter(book_id = book_id, status = True)
+
+        # If copy is available
+        if books.count() > 0:
+            msg = 'success' 
+            # Select the first copy
+            books = books.first()
+            # Set to unavailable
+            books.status = False
+            # User is the borrower
+            books.borrower = request.user
+            # Current date is borrow date
+            books.borrow_date = datetime.date.today()
+            # Save the model object
+            books.save()
+        else:
+            # The following two are possible:
+            #   1. Either no such book is found, or
+            #   2. no such book is available for loan
+            msg = 'failure'
 
+        # Return the message
+        response_data = {
+            'message': msg,
+        }
+        return JsonResponse(response_data)
+
+    # This page is not available for user to view and is just for data exchange
+    else:
+        raise Http404
 
-    return JsonResponse(response_data)
 
-'''
-FILL IN THE BELOW VIEW BY YOURSELF.
-This view will return the issued book.
-You need to accept the book id as argument from a post request.
-You additionally need to complete the returnBook function in the loaned_books.html file
-to make this feature complete
-''' 
 @csrf_exempt
 @login_required
 def returnBookView(request):
-    pass
+    # Books are returned through POST method 
+    if request.method == 'POST': 
+        # Get the desired book copy
+        id = request.POST['bid']
+        book = BookCopy.objects.filter(id = id, status = False).first()
+
+        # If not found, book = None
+        if book != None:
+            # Set status to available
+            book.status = True
+            # Set borrower to none
+            book.borrower = None
+            # Set borrow date to none
+            book.borrow_date = None
+            # Save the object
+            book.save()
+            msg = 'success'
```

Comment:
There must be a validation in the backend when a user is returning the book, to make sure that he has only borrowed the book. Otherwise, a simple POST request will make the BookCopy to be returned, and would set its status as True.